### PR TITLE
Adding stglib

### PR DESCRIPTION
**Pull request by dnowacki-usgs**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml" 
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

**Comment by dnowacki-usgs** [comment]
@conda-forge/staged-recipes Checks are passing and I think this is ready for review. Thank you!

File: recipes/stglib/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python
+    - matplotlib
```
Comment (npavlovikj) [inline recipes/stglib/meta.yaml]: Can this be `matplotlib-base` instead? `matplotlib` provides the Qt backend, but if you don't need the interactive images and just need to render them, then you should use `matplotlib-base`.

File: recipes/stglib/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    - xmltodict
+
+test:
+  imports:
+    - stglib
+
+about:
+  home: http://github.com/dnowacki-usgs/stglib
+  license: CC0
+  license_file: LICENSE.md
```
Comment (npavlovikj) [inline recipes/stglib/meta.yaml]: Please add `license_family: CC`.

File: recipes/stglib/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python
+    - matplotlib
+    - netcdf4
+    - numpy
+    - pandas
+    - pytest
```
Comment (npavlovikj) [inline recipes/stglib/meta.yaml]: Is `pytest` needed since no tests are performed?

File: recipes/stglib/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+test:
+  imports:
+    - stglib
+
+about:
+  home: http://github.com/dnowacki-usgs/stglib
+  license: CC0
+  license_file: LICENSE.md
+  summary: 'Process data from a variety of oceanographic instrumentation'
+
```
Comment (npavlovikj) [inline recipes/stglib/meta.yaml]: Please remove empty line.

File: recipes/stglib/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  noarch: python
+  number: 0
+  script: "{{ PYTHON }} -m pip install . -vv"
+
+requirements:
+  host:
+    - python >=3.6
+    - pip
+  run:
+    - python
```
Comment (npavlovikj) [inline recipes/stglib/meta.yaml]: This should probably also be `python >=3.6`.

**Comment by npavlovikj** [comment]
Thank you for contributing and for the modifications @dnowacki-usgs , the recipe LGTM!

**Pull request merged**